### ISSUE TITLE
ECG Synthesis using MIMIC-IV ECGs

### ISSUE THREAD

**Issue by sanketahegde** [AI4HealthUOL/SSSD-ECG#19]
Hello @juanlopezcode ,

I have been trying to process the MIMIC-IV ECGs to use for training your SSSD-ECG model with the intention of generating ECGs.
I also created multi-labels and ended up with a label length of 20. Currently, there are around 200,000 ECGs which I use for training which is quite large compared to PTB-XL dataset. At the end, I have my training set with `data shape - (220000, 12, 5000)` and `label shape - (220000, 20)`. I already tried training once with these ECGs but the model did not learn properly and also started over-fitting after 6 epochs.

Now, my question is what modifications must I do to the currently available SSSD-ECG model setup to train and generate with MIMIC-IV ECGs?

Any suggestions are appreciated.
Thank you!

**Comment by juanlopezcode** [comment]
Hi, 
I have never trained in the 5000 time step setting, perhaps is too long? I would try first at the 1000 (resample from 500hz to 100hz).
also, not sure how the overfitting was measured or what it represents here. 
I would keep intact the network, and just train on 100Hz, it should already learn a lot because you have a lot of data.

**Comment by sanketahegde** [comment]
Currently, I clean the ECGs to remove any noise and perform lead-wise min-max normalization as part of my preprocessing.

When working with the PTB-XL dataset, I directly used the final data splits which was uploaded online. So, I am unaware of the necessary preprocessing steps.
My follow-up question is what preprocessing steps must be applied on the MIMIC-IV ECGs before feeding into the model? 

Thank you for the suggestions

**Comment by juanlopezcode** [comment]
no preprocessing, just raw waveforms
